src/Processors/Formats/IInputFormat.cpp

```diff
@@ -1,10 +1,93 @@
 #include <Processors/Formats/IInputFormat.h>
+#include <Processors/Formats/Impl/CSVRowOutputFormat.h>
 #include <IO/ReadBuffer.h>
+#include <IO/WriteHelpers.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypesNumber.h>
 
 
 namespace DB
 {
 
+InputFormatErrorsLogger::InputFormatErrorsLogger(
+    Context::ApplicationType app_type,
+    const String & user_files_path,
+    String & path_in_setting,
+    bool is_changed,
+    FormatSettings::ErrorsOutputFormat output_format,
+    const String & database_,
+    const String & table_)
+    : database(database_), table(table_)
+{
+    if (app_type == Context::ApplicationType::SERVER)
+    {
+        trimLeft(path_in_setting, '/');
+    }
+    else if (!is_changed)
+    {
+        path_in_setting = "/tmp/" + path_in_setting;
+    }
+    errors_file_path = user_files_path + path_in_setting;
+    if (is_changed)
+    {
+        while (fs::exists(errors_file_path))
+        {
+            errors_file_path += "_new";
+        }
+    }
+    write_buf = std::make_shared<WriteBufferFromFile>(errors_file_path);
+
+    Block header{
+        {std::make_shared<DataTypeString>(), "time"},
+        {std::make_shared<DataTypeString>(), "database"},
+        {std::make_shared<DataTypeString>(), "table"},
+        {std::make_shared<DataTypeUInt32>(), "offset"},
+        {std::make_shared<DataTypeString>(), "reason"},
+        {std::make_shared<DataTypeString>(), "raw_data"}};
+    FormatSettings format_settings;
+    RowOutputFormatParams out_params;
+
+    if (output_format == FormatSettings::ErrorsOutputFormat::CSV)
+        writer = std::make_shared<CSVRowOutputFormat>(*write_buf, header, false, false, out_params, format_settings);
```

Comment:
We can call `context.getOutputFormat(...)` and it will return output format according to format name, so we can use here any supported format, not only CSV